src/main/java/org/gestern/gringotts/AccountInventory.java

```diff
@@ -49,10 +49,10 @@ public long add(long value) {
 
         // try denominations from largest to smallest
         for (Denomination denomination : Configuration.CONF.getCurrency().getDenominations()) {
-            if (denomination.getValue() <= remaining) {
+            if (denomination.getValue() <= remaining && denomination.getValue() > 0) {
```

Comment:
I made this change to make it clear that we skip this block if the denomination's value is 0; we were already doing that with the ternary and while together, but this makes it more obvious.